modules/custom/activity_viewer/src/Plugin/views/filter/ActivityPostVisibilityAccess.php

```diff
@@ -205,7 +224,11 @@ public function query():void {
     $post_access = new Condition('AND');
     $post_access->condition('activity__field_activity_entity.field_activity_entity_target_type', 'post');
 
-    if (!$account->hasPermission('bypass group access')) {
+    // Get group from url-parameter.
+    $group = $this->routeMatch->getParameter('group');
```

Comment:
you are lucky that https://github.com/goalgorilla/open_social/pull/4118 is not merged yet, it brings a current group service that could have been used. :)